doc/manual/noteworthy-differences.rst

```diff
@@ -251,3 +251,75 @@ Noteworthy differences from Python
   returns a new random number every time it is invoked without argument. On the
   other hand, the function ``g(x=[1,2]) = push!(x,3)`` returns ``[1,2,3]`` every
   time it is called as ``g()``.
+
+
+Noteworthy differences from C/C++
+----------------------------------
+
+- Julia arrays are indexed with square brackets, and can have more than one
+  dimension ``A[i,j]``.
+- In Julia, indexing of arrays, strings, etc. is 1-based not 0-based.
+- Julia arrays are assigned by reference. After ``A=B``, changing elements of
+  ``B`` will modify ``A`` as well.
+- Julia values are passed and assigned by reference. If a function modifies an
+  array, the changes will be visible in the caller.
+- In Julia, literal numbers without a decimal point (such as ``42``) create signed
+  integers, of type Int, but literals too large to fit in the machine word size
+  will automatically be promoted to a larger size type, such as Int64 (if Int is Int32),
+  Int128, or the arbitrarily large BigInt type.
+  There are no numeric literal suffixes, such as L, LL, U, UL, ULL to indicate unsigned
+  and/or signed vs. unsigned.
+  Decimal literals are always signed, and hexadecimal literals (which start with 0x like C/C++),
+  are unsigned (except if they are >128 bits, in which case they become signed BigInt type).
+  Hexadecimal literals also, unlike C/C++/Java and unlike decimal literals in Julia,
+  have a type based on the *length* of the literal, including leading 0s.  For example,
+  ``0x0`` and ``0x00`` have type UInt8, ``0x000`` and ``0x0000`` have type UInt16, then
+  literals with 5 to 8 hex digits have type UInt32, 9 to 16 hex digits type UInt64, and more than
+  16 hex digits end up a signed BigInt type.  This needs to be taken into account when defining
+  hexadecimal masks, for example `~0xf` is very different from `~0x000f`.
+  Unlike either decimal integer or hexadecimal literals, floating point literals are always
+  64-bits, Float64, and don't get promoted to the BigFloat type.  This is closer in
+  behavior to C/C++.
+- String literals can be delimited with either `"`  or `"""`, `"""` delimited literals can contain `"` characters without
+  quoting it like `"\\\""`
+  They can also have string interpolation, indicated by $variablename or $(expression),
```

Comment:
It's not clear here whether "They" refers to only `"""` delimited strings. I'd say something like "String literals can have values of other variables or expressions interpolated into them, indicated by..."

Should also put ``` ` ``` around `$variablename` and `$(expression)`